### PR TITLE
haveSame is suboptimal, fix it as well as the name

### PR DESCRIPTION
**Pull request by anfernee**
**What this PR does / why we need it**: Make the util function simpler, also faster by reducing a for loop.

**Special notes for your reviewer**:

**Release note**:
```release-note
None
```

### CODE REVIEW

**Comment by anfernee** [comment]
cc @timothysc @resouer

**Comment by anfernee** [comment]
cc @bsalamat

File: plugin/pkg/scheduler/algorithm/predicates/predicates.go [hunk 1]
```diff
@@ -911,20 +911,18 @@ func PodFitsHostPorts(pod *v1.Pod, meta algorithm.PredicateMetadata, nodeInfo *s
 }
 
 // search two arrays and return true if they have at least one common element; return false otherwise
-func haveSame(a1, a2 []string) bool {
+func isOverlapped(a1, a2 []string) bool {
```
Comment (bsalamat) [inline plugin/pkg/scheduler/algorithm/predicates/predicates.go]: s/isOverlapped/haveOverlap

File: plugin/pkg/scheduler/algorithm/predicates/predicates.go [hunk 2]
```diff
@@ -911,20 +911,18 @@ func PodFitsHostPorts(pod *v1.Pod, meta algorithm.PredicateMetadata, nodeInfo *s
...
 	m := map[string]int{}
```
Comment (bsalamat) [inline plugin/pkg/scheduler/algorithm/predicates/predicates.go]: Now that you have improved it, please make the map value type bool:
```go
map[string]bool{}
```

**Review by bsalamat — COMMENTED** [review]
Thanks, @anfernee! Looks good. Just a couple minor comments.

**Pull request merged**